doc/developer/design/20231110_aws_connections.md

```diff
@@ -0,0 +1,164 @@
+Creating an AWS connection with credentials should be quite straightforward. The SQL would look like,
+```
+CREATE SECRET aws_secret_key = 'secret';
+
+CREATE CONNECTION aws_conn TO AWS (
+  ACCESS KEY ID = 'access_key',
+  SECRET ACCESS KEY = SECRET aws_secret_key
+);
+
+```
+The `aws_conn` is now available to be used in supported queries.
+
+### AWS IAM AssumeRole
+Users will need to first create a role on their end, say, `MaterializeConn` and note down it's arn, e.g: `arn:aws:iam::001234567890:role/MaterializeConn`.
+
+User then creates a connection in Materialize with that arn.
+```
+CREATE CONNECTION aws_conn TO AWS (
+  ASSUME ROLE ARN = 'arn:aws:iam::001234567890:role/MaterializeConn'
+);
+```
+On Materialize's end we generate an External ID and create an entry in a new `mz_aws_connections` table.
+Users should be able to get the external_id and the principal by querying this table.
+```
+SELECT principal, external_id
+FROM mz_aws_connections aws
+JOIN mz_connections c ON aws.id = c.id
+WHERE c.name = 'aws_conn';
+```
+
+Users should then be able to add a trust policy in their AWS account to give access to Materialize.
+```
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "<principal>"
+      },
+      "Action": "sts:AssumeRole",
+      "Condition": {
+        "StringEquals": {
+          "sts:ExternalId": "<external_id>"
+        }
+      }
+    }
+  ]
+}
+```
+
+## Implementation
+We already have some [existing code](https://github.com/MaterializeInc/materialize/blob/v0.77.1/src/storage-types/src/connections/aws.rs), which can be re-used to
+handle the AssumeRole scenario.
+
+In the parser we already support `ROLE ARN` option which can be renamed to `ASSUME ROLE ARN`
+to better reflect the usage. We will need to add an `ASSUME ROLE SESSION NAME` option as well.
+
+The AWS Connection create statement with all the options will now look like:
+```sql
+CREATE CONNECTION <connector_name> TO AWS (
+  ACCESS KEY ID [[=] <value>], -- existing
+  SECRET ACCESS KEY [[=] SECRET <value>], -- existing
+  TOKEN [[=] SECRET <value>], -- existing
+  ASSUME ROLE ARN [[=] <value>], -- rename existing from ROLE ARN
+  ASSUME ROLE SESSION NAME [[=] <value>], -- new option
+  ENDPOINT [[=] <value>], -- existing
+  REGION [[=] <value>] -- existing
+);
+```
+
+Users should be able to provide either the `ACCESS KEY` options or the `ASSUME ROLE` options,
+but not both.
+
+#### External ID and principal
+We already have external ID prefix provided in the catalog state.
+The connection ID will be appended to the external ID prefix to get the complete
+External ID and then stored in the new `mz_aws_connections` table for that connection ID.
+
+For principal, we'll create one global role which all environmentd/clusterd-s can assume.
+This will add some good restrictions on who could assume the customers role and also
+allow for migration between environment clusters.
```

Comment:
This sounds good. We should specify (and discuss) the name of that role here!

Reply:
Added @jubrad's suggestion `MaterializeCloudConnectionAssumeRole` above.